Adding external registry configuration capability to registry tests

izaac opened the pull request:
## Issue: <!-- link the issue or issues this PR resolves here -->
Private registries tests create new registries only for running the tests on clusters. 
 
## Problem

Creating new registries during test setup take time between tests runs. If there are multiple runs needed.

 
## Solution
This adds the capability of setup configuration using existing registries. This is mainly for downstream clusters.
The local cluster is configuring during the setup build with Corral. And there's a [change already in](https://github.com/rancherlabs/corral-packages/pull/9) corral packages for this.

This adds support for both type of registries. Auth and No Auth.


### Automated Testing
Testing adding existing registries flags with the new existing registries configuration keys.
Test run without specifying registries. Test keep creating new registries for the setup as expected.

## QA Testing Considerations
None
 
### Regressions Considerations
Just the private registries automated tests are impacted by the change.


### New configuration keys structure:

```yaml
registries:
  existingAuthRegistry:
    password: <password>
    url: <auth registry url>
    username: <auth registry username>
  existingNoAuthRegistry: <no auth registry url>
```

caliskanugur commented on tests/v2/validation/registries/config.go:
```diff
@@ -1,7 +1,18 @@
-const RegistriesConfigKey = "registries"
+const (
+	RegistriesConfigKey 	      = "registries"
+	ExistingAuthRegistryConfigKey =  "existingAuthRegistryConfig"
```
This one probably needs `go.fmt` run again

caliskanugur commented on tests/v2/validation/registries/config.go:
```diff
@@ -1,7 +1,18 @@
-	RegistryConfigNames []string `json:"registryConfigNames" yaml:"registryConfigNames" default:"[]"`
+	RegistryConfigNames      []string `json:"registryConfigNames" yaml:"registryConfigNames" default:"[]"`
+	ExistingNoAuthRegistryURL string `json:"existingNoAuthRegistry" yaml:"existingNoAuthRegistry" default:""`
+	ExistingAuthRegistryInfo  *ExistingAuthRegistryConfig `json:"existingAuthRegistry" yaml:"existingAuthRegistry" default:""`
```
I think we don't need the `ExistingAuthRegistryConfigKey` constant above. Since the `ExistingAuthRegistryConfig` structure is used here, the key to this struct gonna be `existingAuthRegistry`, but not from `ExistingAuthRegistryConfigKey`; from the defined tag here `json:"existingAuthRegistry" yaml:"existingAuthRegistry" default:"`

izaac replied:
Thanks for the feedback Ugur, I'll make the changes and test it out. 🙏🏽

igomez06 replied:
agreed

caliskanugur reviewed (commented):
I've got only a nit, looks good to me

igomez06 reviewed (commented):
same nit as Ugur

izaac commented:
On the latest change I fixed the local cluster logic for the different Rancher install scenarios. This was found while doing the pipeline runs.


- New Rancher with Registry (corral)
- New Rancher without Registry (corral)
- Existing Rancher with Registry (no corral)
- Existing Rancher without Registry

I've added a temporary timeout for the RKE1 provisioning due to https://github.com/rancher/qa-tasks/issues/734
I'll update once that's done.

igomez06 reviewed (dismissed):
Looks good, are we going to wait for the timeout fix PR? I forgot, does the need a backport as well?

izaac commented:
> Looks good, are we going to wait for the timeout fix PR? I forgot, does the need a backport as well?

Hey Israel, I'd rather wait for the timeout improvement.
I will open a backport once I update it and get the approvals if that sounds good.

izaac commented:
Backport: https://github.com/rancher/rancher/pull/41596

slickwarren commented:
there are some merge conflicts, from my latest PR from what it looks like

izaac commented:
Rebased, all other comments were resolved @slickwarren as the new framework timeout is in too.
